File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqs = {}
        elements = [[] for i in range(len(nums) + 1)]
        for num in nums:
            freqs[num] = 1 + freqs.get(num, 0)
        
        for num, freq in freqs.items():
            elements[freq].append(num)
        
        sol = []
        for i in range(len(elements) - 1, 0, -1):
            for num in elements[i]:
                sol.append(num)
                if len(sol) == k:
                    return sol
        
        

File: Data Structures & Algorithms/house-robber/submission-2.java
class Solution {
    public int rob(int[] nums) {
        int max = nums[0];
        int[] dp = new int[nums.length];
        dp[0] = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (i == 1) {
                dp[1] = Math.max(nums[0], nums[1]);
            } else {
                dp[i] = Math.max(dp[i - 2] + nums[i], dp[i - 1]);
            }
            max = Math.max(max, dp[i]);
        }
        return max;
    }
}
